rscripts/YAL5-codon-pairs.R

```diff
@@ -0,0 +1,1455 @@
+#' @param feature_of_interest character, each incidence of the feature will be
+#' extracted from transcript_info_tibble
+#' @param expand_width integer which provides the number of positions on each
+#' side of the feature of interest to include in the window
+#'
+#' @return a list of tibbles, one tibble for each occurrence of the feature of
+#' interest with an expanded window around the feature
+#'
+#' @example
+#'
+#' gff_df <- readGFFAsDf(here::here("..",
+#'                                  "example-datasets",
+#'                                  "simulated",
+#'                                  "mok",
+#'                                  "annotation",
+#'                                  "Scer_YAL_5genes_w_250utrs.gff3"))
+#'
+#' ExpandFeatureRegion(codon_pos_pair_i200,
+#'                     gene_names,
+#'                     dataset = "Mok-simYAL5",
+#'                     hd_file = here::here("Mok-simYAL5",
+#'                                          "output",
+#'                                          "A",
+#'                                          "A.h5"),
+#'                     gff_df,
+#'                     min_read_length = 10,
+#'                     snapdisp = 0L,
+#'                     asite_disp_path = here::here(
+#'                       "data",
+#'                       "yeast_standard_asite_disp_length.txt"),
+#'                     filter_for_frame = TRUE,
+#'                     feature_of_interest = "TCC AAG",
+#'                     expand_width = 5L
+#'                     )
+#'
+#' @export
+ExpandFeatureRegion <- function(codon_pos_pair_i200, gene_names, dataset,
+                                hd_file, gff_df, min_read_length, snapdisp,
+                                asite_disp_path, filter_for_frame,
+                                feature_of_interest, expand_width) {
+
+  # fetch the assigned read counts for genes in gene_names
+  gene_pos_codon_counts <- AddCodonNamesToCodonPosCounts(codon_pos_pair_i200,
+                                                         gene_names,
+                                                         dataset,
+                                                         hd_file,
+                                                         min_read_length,
+                                                         snapdisp,
+                                                         asite_disp_path,
+                                                         filter_for_frame)
+
+  # takes gene_pos_codon_counts as inputs and select for positions
+  # on separate genes
+  AllGeneInterestingFeatures <- function(codon_pos_pair_i200,
+                                         gene, gene_names, dataset, hd_file,
+                                         min_read_length, feature_of_interest,
+                                         gene_pos_codon_counts,
+                                         gff_df) {
+
+    # to indicate which gene is being processed
+    print(gene)
```

Comment:
remove from final code, good for testing but makes the console cluttered for user